### PR TITLE
Real-time data: provide the data connector JS via Connectors

### PR DESCRIPTION
**Pull request by nadzpogi**
relates to xibosignageltd/xibo-private#753

### CODE REVIEW

File: lib/Event/DataConnectorSourceRequestEvent.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ */
+class DataConnectorSourceRequestEvent extends Event implements DataConnectorSourceProviderInterface
+{
+    public static $NAME = 'data.connector.source.request';
+
+    /**
+     * @var array
+     */
+    private $dataConnectorSources = [
+        ['id' => 'user_defined', 'name' => 'User Defined JavaScript']
```
Comment (dasgarner) [inline lib/Event/DataConnectorSourceRequestEvent.php:41]: Can we translate the name?

File: views/dataset-data-connector-page.twig [hunk 2]
```diff
@@ -67,7 +67,7 @@
                                 </div>
                             </div>
 
-                            {{ forms.button("Save"|trans, "submit", null, null, null, "btn-success") }}
+                            {{ forms.button("Save"|trans, "submit", null, null, null, "btn-success " ~ (dataSet.dataConnectorSource != 'user_defined' ? 'disabled' : ''), disabled) }}
```
Comment (dasgarner) [inline views/dataset-data-connector-page.twig:70]: Should we hide it as well?

**Review by dasgarner — CHANGES_REQUESTED** [review]
This is great - I only have a very minor comment regarding translating.

File: lib/Event/DataConnectorSourceRequestEvent.php (hunk 1 above)
Comment (nadzpogi) [reply lib/Event/DataConnectorSourceRequestEvent.php:41]: Noted. Sorry, the translation slipped my mind. 😅

File: views/dataset-data-connector-page.twig (hunk 2 above)
Comment (nadzpogi) [reply views/dataset-data-connector-page.twig:70]: Oh, it's already within the div that gets hidden if the source is not user-defined, so there's no need to explicitly hide the button within that div.

But I forgot to remove the disabled parameter (it was a test to see if it works), so I'll remove that one.
` ''), disabled) }}`

**Review by dasgarner — APPROVED** [review]

**Pull request merged**